### ISSUE TITLE
Feature Early Stopping

### ISSUE THREAD

**Issue by AnantaCoder** [Sree14hari/EPOQ#67]
Currently, the script runs for all specific epochs.
Contribution: Modify the training loop in  to monitor validation loss. If loss doesn't improve for n epochs (patience), stop training and save the best model.

Why add it?

Saves Time: Training stops automatically when the model stops learning.
Prevents Overfitting: It stops the model before it starts memorizing the training data and performing worse on unseen data.
